Lib/random.py

```diff
@@ -379,6 +383,16 @@ def sample(self, population, k):
             population = tuple(population)
         if not isinstance(population, _Sequence):
             raise TypeError("Population must be a sequence.  For dicts or sets, use sorted(d).")
+        if weights is not None:
+            cum_weights = list(_accumulate(weights))
```

Comment:
Hi, failure mode when `len(weights) != len(population)` is weird at the moment:

```
>>> from random import sample
>>> dealt = sample(['tens', 'low cards'], weights=[16, 32, 4], k=20)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/remi/src/cpython/Lib/random.py", line 395, in sample
    return [population[bisect(cum_weights, s)] for s in selections]
  File "/Users/remi/src/cpython/Lib/random.py", line 395, in <listcomp>
    return [population[bisect(cum_weights, s)] for s in selections]
IndexError: list index out of range
>>> dealt = sample(['tens', 'low cards'], weights=[16], k=20)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/remi/src/cpython/Lib/random.py", line 393, in sample
    selections = sample(range(total), k=k)
  File "/Users/remi/src/cpython/Lib/random.py", line 399, in sample
    raise ValueError("Sample larger than population or is negative")
ValueError: Sample larger than population or is negative
```


```suggestion
            cum_weights = list(_accumulate(weights))
            if len(cum_weights) != len(population):
                raise ValueError('The number of weights does not match the population')
```

Reply:
Add the test. Thanks.